### DISCUSSION TITLE
unusual hydration error on error page

### DISCUSSION THREAD

**Discussion by cadentic** [vercel/next.js#40403] [Help] [answered]
hi

I have a very small issue I guess at this point ie. proper error handling. I am a little new in the js paradigm. 
![Capture_on_error_page](https://user-images.githubusercontent.com/55697529/189377884-6ff5f093-33f6-47a7-898e-1145e3068c72.JPG)

```
import Head from 'next/Head';
import Copyright from '../src/Copyright';
import style from "../styles/Home.module.css";

function Error({ statusCode }) {
    return (
       
        <main className={style.main}>
       <Head>
  <title>Create Next App</title>
  <link rel="icon" href="/favicon.ico" />
</Head>
  <h1 className={style.title}>
    Welcome to <a href="https://z">
      some text</a> integrated with{" "}
                <a href="/">some text</a>
                
  </h1>
            <p className={style.description}>
    Get started by searching {" "}
    <code className={style.code}>
      pages/ or new keyword </code>
  </p>
            <p className={style.description}> 
    {statusCode
        ? ` An error ${statusCode} occurred on server `
        : 'An error occurred on client'}
</p>
        <Copyright />
</main>
    );
}

Error.getInitialProps = ({ res, err }) => {
    const statusCode = res ? res.statusCode : err ? err.statusCode : 404
    return { statusCode }
}

export default Error
```

**Comment by icyJoseph** [comment]
How are you triggering the error? Got some link or repository to look at? 

At least, where is the mismatch happening? The browser console contains more information. Perhaps the issue is with status code, and that the contents of that `p` tag diverge between server and client.

**Reply by cadentic** [reply]
> How are you triggering the error? Got some link or repository to look at?
> 
> At least, where is the mismatch happening? The browser console contains more information. Perhaps the issue is with status code, and that the contents of that `p` tag diverge between server and client.

[it is the same repo -- my repo](https://github.com/cadentic/strapi-webdev/blob/main/zettadaten_fe_next17/pages/_error.js)

**Reply by icyJoseph** [reply]
Alright, and what steps do you follow to trigger the error?

**Comment by cadentic** [comment]
I did not trigger any error from the pages but if I am visiting any non-existing pages say cat.js it is showing the hydration error. 
it should trigger only 404.js and for other errors eg. if strapi connectivity fails, it should go to _error.js  instead of hydration error.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Your issue is that the 404 page is using invalid HTML nesting. You have `p` tags nested within each other, https://caninclude.glitch.me/caninclude?child=p&parent=p:


https://github.com/cadentic/strapi-webdev/blob/9b13729fa6f6171b8b9a9acbd86d1b641b63392a/zettadaten_fe_next17/pages/404.js#L20-L28

Shows the problem, `Typography` is rendering a `p` tag, and then directly inside you have another `p` tag and that's causing invalid DOM nesting which produces different HTML on the server and on the client and that creates a hydration error. You also have an issue with `p` as child to `code`, https://caninclude.glitch.me/caninclude?child=p&parent=code.

- For clarity, the `p` tag directly under `Typography` is problematic.
- `p` inside the last `code` tag is also problematic

```jsx
<Typography variant="body2">
        <p className={style.description}>
          Get started by searching {" "}
        </p>  
        <code className={style.code}> pages/ or new keyword ; </code>
        <code className={style.code}>  We are greateful to know that you have achieved page not found url
          <p className={style.description}>   we will respond to it quickly but till then try to visit something else </p>
        </code>
</Typography>
```

Please do pay attention to the console for errors, the messages there are very useful.

**Reply by cadentic** [reply]
awesome but how if we are using ```<Error />``` tag/components from ```next/error``` it will not redirect to error page but will show error in same page so how can we forward error status code to ```_error.js``` page?

**Reply by cadentic** [reply]
thanks for extending your hand.
